x/emissions/keeper/keeper.go

```diff
@@ -826,6 +835,16 @@ func (k *Keeper) AppendInference(ctx context.Context, topicId TopicId, nonce typ
 		newInferences.Inferences = append(newInferences.Inferences[:lowScoreIndex], newInferences.Inferences[lowScoreIndex+1:]...)
 		newInferences.Inferences = append(newInferences.Inferences, inference)
 		return k.allInferences.Set(ctx, key, newInferences)
+	} else {
+		// Update EMA score using previous active set score at topic-defined quantile among the last active set of inferers
+		topic, err := k.GetTopic(ctx, topicId)
+		if err != nil {
+			return err
+		}
+		err = k.CalcAndSaveInfererScoreEmaWithLastSavedTopicQuantile(ctx, topic, block, inference.Inferer)
+		if err != nil {
+			return err
+		}
```

Comment:
We are not saving the inference of this inferrer anywhere; is that correct?

Additionally, I realized that in the `AppendInference` function, we are only saving the inferences in the KV store `allInferences` and not in `inferences`, which can affect some queries that use it.

I think we should use the `InsertInferences` method of the keeper.

Reply:
fixed both